airflow/api/common/mark_tasks.py

```diff
@@ -164,7 +164,7 @@ def get_all_dag_task_query(
     qry_dag = select(TaskInstance).where(
         TaskInstance.dag_id == dag.dag_id,
         TaskInstance.run_id.in_(run_ids),
-        TaskInstance.ti_selector_condition(task_ids),
+        TaskInstance.ti_selector_condition(task_ids, session=session),
```

Comment:
It’s quite weird we need to pass in `session` here. Is it possible to do this with some SQL JOIN and/or subquery magic?

Reply:
Thanks for the review!
Removal of session won't affect anything, I did this because I had to comment out the @provide_session in order to debug.  Here, session: Session = NEW_SESSION, it already has a default supplied by the decorator.

Reply:
I've made the changes. If there are any other modifications needed, please let me know so I can include them all in a single commit

Reply:
The session needs to be a `kwarg` of `ti_selector_condition` since the entire logic of filtering revolves around being able to filter out using SQL queries. If the `session` kwarg is not passed the function is decorated by `@provide_session,` which will handle session creation.
One of the tests failed once I removed the session kwarg because of inconsistency between a `taskinstance` and `session,` the `@provide_session` created a new session which lead to this inconsistency.
But if the session is persisted as is shown in the PR, this inconsistency would not occur.

Reply:
My point is that the method _should not_ do a fetch in the first place. The method calculates filtering parameters for a query. Using another query to do that is not a generally preferred approach since it makes it difficult to manage where queries happen. It would be best if the multiple queries can be combined into one with JOIN or other SQL constructs, or, if that is not viable, the queries should at least happen in the same function, not in different levels in the call stack.

Reply:
My apologies @uranusjr, you are absolutely correct. I should have done that from the very beginning. I've completely removed any dependency on `session` and used simple `subquery` to filter out the `TaskMap` objects